realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProcessor.java

```diff
@@ -245,4 +259,32 @@ private boolean createMediator(String simpleModuleName, Set<ClassMetaData> modul
 
         return true;
     }
+
+    // Because library classes are processed separately, there is no guarantee
+    // that this method can see all of the classes necessary to completely validate
+    // all of the backlinks.  If it can find the fully-qualified class, though,
+    // and prove that the class either does not contain the necessary field, or
+    // that it does contain the field, but the field is of the wrong type, it can
+    // catch the error at compile time.
+    // Give all failure messages before failing
+    private boolean validateBacklinks() {
+        boolean allValid = true;
+
+        Map<String, ClassMetaData> realmClasses = new HashMap<String, ClassMetaData>(classesToValidate.size());
+        for (ClassMetaData classData: classesToValidate) {
+            realmClasses.put(classData.getFullyQualifiedClassName(), classData);
+        }
+
+        for (Backlink backlink: backlinksToValidate) {
+            ClassMetaData klass = realmClasses.get(backlink.getSourceClass());
+
+            // If the class is not here it might be part of some other compilation unit.
+            if (klass == null) { continue; }
```

Comment:
Yup, we need to do both, but catching as much as possible on compile time make a better user experience. But fine doing it in another PR 👍